### PR TITLE
Calendar notif

### PR DESCRIPTION
**Pull request by mmoqui**
Three things are refactored and fixed:
- WYSIWYG: the WYSIWYG editor configuration path value in the WYSIWYG settings is now parsed according to its starting scheme (/ or http or nothing)
- Calendar Engine: the notification to the attendees is now refined. When the state of an attendee changes, the event isn't anymore considered as updated by itself; the attendees are then just notified about the attendance change. Any change in an occurrence of a recurrent event triggers a notification about only this occurrence (or about all the occurrences since that occurrence). Any change in the attendance of a given occurrence triggers also a notification about that attendance change.
- Persistence Engine: the way the creation date and author, and the update date and author are treated is refactored. The memorization of an explicit change in those attributes of an entity is now performed in an implementation of PersistenceOperation, a new class that represents a given persistence operation in the Persistence Engine. If no explicit change was performed in those attributes, then the author is considered as the user behind the current request and the date is set at the date and time at which the persistence operation is performed. In the JPA-based implementation of the Silverpeas Persistence API, two concretes classes extends the abstract class PersistenceOperation: JpaUpdateOperation and JpaPersistOperation. They are directly used by SilverpeasJpaEntity (a base class for all persistent entities that used JPA for their persistence with the Persistence Engine) and they are obtained from the OperationContext object in the current thread. Another change: the creation attributes and the update attributes of a persistence entity are now decorrelated: a change in the creation attributes doesn't set also the update attributes; so with such a change, all entities coming from an import can be persisted or updated with their own creation and update attribute values (as they are valued in the import source).  

Don't forget to merge also the PR https://github.com/Silverpeas/Silverpeas-Components/pull/556

### CODE REVIEW

File: core-api/src/main/java/org/silverpeas/core/persistence/datasource/OperationContext.java [hunk 1]
```diff
@@ -53,6 +55,8 @@
 
   // The user
   private User user = null;
+  private List<PersistenceOperation> persistenceOperations = new ArrayList<>();
+  private List<SilverpeasJpaEntity> entities = new ArrayList<>();
```
Comment (silveruser) [inline core-api/src/main/java/org/silverpeas/core/persistence/datasource/OperationContext.java]: MAJOR Remove this unused "entities" private field. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1068)

File: core-api/src/main/java/org/silverpeas/core/persistence/datasource/model/jpa/JpaEntityReflection.java [hunk 2]
```diff
@@ -37,69 +35,22 @@
  */
 public class JpaEntityReflection {
 
-  private static Method createDateMethod = null;
-  private static Method lastUpdateDateMethod = null;
-  private static Field createdBySetManuallyField = null;
-  private static Field lastUpdatedBySetManuallyField = null;
+  private static Field creatorSetManuallyField = null;
```
Comment (silveruser) [inline core-api/src/main/java/org/silverpeas/core/persistence/datasource/model/jpa/JpaEntityReflection.java]: MAJOR Remove this unused "creatorSetManuallyField" private field. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1068)

File: core-api/src/main/java/org/silverpeas/core/persistence/datasource/model/jpa/JpaEntityReflection.java [hunk 3]
```diff
@@ -37,69 +35,22 @@
...
+  private static Field lastUpdaterSetManuallyField = null;
```
Comment (silveruser) [inline core-api/src/main/java/org/silverpeas/core/persistence/datasource/model/jpa/JpaEntityReflection.java]: MAJOR Remove this unused "lastUpdaterSetManuallyField" private field. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1068)

**Comment by silveruser** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [Attendee.java#L26](https://github.com/Silverpeas/Silverpeas-Core/blob/52dcf4a4ea7c6dfee5f43e164911b045a3aea36b/core-api/src/main/java/org/silverpeas/core/calendar/Attendee.java#L26): Remove this unused import 'org.apache.commons.lang3.builder.HashCodeBuilder'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**